server/src-lib/Hasura/RQL/Types/CustomTypes.hs

```diff
@@ -252,17 +258,17 @@ fieldTypeToScalarType = \case
            | _stdName == boolScalar   -> PGBoolean
            | otherwise                -> PGJSON
 
-type AnnotatedObjectType =
-  ObjectTypeDefinition (G.GType, AnnotatedObjectFieldType) TableInfo PGColumnInfo
+type AnnotatedObjectType b =
+  ObjectTypeDefinition (G.GType, AnnotatedObjectFieldType b) (TableInfo b) (ColumnInfo b)
 
-type AnnotatedObjects = Map.HashMap G.Name AnnotatedObjectType
+type AnnotatedObjects b = Map.HashMap G.Name (AnnotatedObjectType b)
 
-data AnnotatedCustomTypes
+data AnnotatedCustomTypes (b :: Backend)
```

Comment:
CustomTypes span all the sources, i.e, their definitions can use types from any source and relationships to any source, so the backend parameter is not appropriate here.

Reply:
In `AnnotatedObjectType`, is there a requirement that the `AnnotatedObjectFieldType` comes from the same backend as the specified `TableInfo` and `ColumnInfo`?